parsel/csstranslator.py

```diff
@@ -89,6 +101,116 @@ def xpath_text_simple_pseudo_element(self, xpath):
         """Support selecting text nodes using ::text pseudo-element"""
         return XPathExpr.from_xpath(xpath, textnode=True)
 
+    def xpath_first_of_pseudo(self, xpath):
```

Comment:
I'm puzzled on how this works -- shouldn't the functions match the pattern `xpath_%s_simple_pseudo_element` or `xpath_%s_functional_pseudo_element`, like dispatched by the `xpath_pseudo_element()` method above?
Who calls these methods?